src/components/Genres/Genres.spec.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import Genres from "./Genres";
+
+describe("Genres Component", () => {
+  const genresMock = ["All", "Documentary", "Comedy", "Horror", "Crime"];
+  const selectedGenreMock = "All";
+  const onSelectMock = jest.fn();
+
+  const GenresMock = (
+    genres: string[],
+    selectedGenre: string,
+    onSelect: jest.Mock
+  ) => {
+    render(
+      <Genres
+        genres={genres}
+        selectedGenre={selectedGenre}
+        onSelect={onSelect}
+      />
+    );
+  };
+
+  it("Should render all genres", () => {
+    GenresMock(genresMock, selectedGenreMock, onSelectMock);
```

Comment:
You are passing the same exact mocked values in each test, which can lead to coupling of the tests. For example, what if some test modifies `genresMock` array? Then the tests that follow it could get unexpected behavior. It is better to initialize an array each time, for example by adding the spread operator. 
For mocks, you should clear all mocks after each test to make sure `onSelectMock` doesn't hold the results of previous tests.

Reply:
I can see that you've added `afterEach` with `jest.clearAllMocks` which is good, but I would also make sure that you are using a separate genres array in each test:
```
...
    render(
      <Genres
        genres={[...genres]}
...
```
It is really important to keep tests isolated and independent of each other.